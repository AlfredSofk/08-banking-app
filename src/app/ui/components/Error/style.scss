.error-message {
    background-color: #ffe6e6;
    color: #b71c1c;
    border: 1px solid #f44336;
    border-radius: 8px;
    padding: 16px;
    max-width: 600px;
    margin: 20px auto;
    box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 12px;
  
    &__header {
      margin-bottom: 12px;
    }
  
    &__title {
      font-size: 24px;
      font-weight: bold;
      margin: 0;
      color: #b71c1c;
    }
  
    &__body {
      font-size: 16px;
      line-height: 1.5;
    }
  
    &__description {
      margin: 0 0 10px;
    }
  
    &__details {
      background-color: #fff8f8;
      border-left: 4px solid #f44336;
      padding: 10px;
      border-radius: 4px;
      font-size: 14px;
      color: #d32f2f;
  
      summary {
        font-weight: bold;
        cursor: pointer;
        margin-bottom: 8px;
      }
    }
  }
  