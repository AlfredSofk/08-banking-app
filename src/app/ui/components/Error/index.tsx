import React from "react";
import "./style.scss";

interface ErrorMessageProps {
  title?: string; // Título del mensaje de error
  description?: string; // Descripción del error
  errorDetails?: string; // Detalles adicionales del error
}

export const ErrorScreen: React.FC<ErrorMessageProps> = ({
  title = "Something went wrong",
  description = "An unexpected error occurred. Please try again later.",
  errorDetails,
}) => {
  return (
    <section
      className="error-message"
      role="alert"
      aria-live="assertive"
      aria-atomic="true"
    >
      <header className="error-message__header">
        <h1 className="error-message__title">{title}</h1>
      </header>
      <div className="error-message__body">
        <p className="error-message__description">{description}</p>
        {errorDetails && (
          <details className="error-message__details">
            <summary className="error-message__summary">More details</summary>
            <p>{errorDetails}</p>
          </details>
        )}
      </div>
    </section>
  );
};