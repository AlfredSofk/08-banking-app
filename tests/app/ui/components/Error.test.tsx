import { render, screen } from '@testing-library/react';
import { ErrorScreen } from '../../../../src/app/ui/components/Error/index';
import React from 'react';



describe('Test para el componente ErrorScreen', () => {
    test('match snapshot', () => {
        const { asFragment } = render(<ErrorScreen />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('renderiza el título y la descripción por defecto', () => {
        render(<ErrorScreen />);
    
        // Verificar que el título y la descripción por defecto están presentes
        expect(screen.getByRole('alert')).toBeInTheDocument();
        expect(screen.getByRole('alert')).toHaveAttribute('aria-live', 'assertive');
        expect(screen.getByRole('alert')).toHaveAttribute('aria-atomic', 'true');
    
        expect(screen.getByText('Something went wrong')).toBeInTheDocument();
        expect(
          screen.getByText(
            'An unexpected error occurred. Please try again later.'
          )
        ).toBeInTheDocument();
      });
    
      test('renderiza el título y la descripción personalizados', () => {
        render(
          <ErrorScreen
            title="Custom Error"
            description="This is a custom error description."
          />
        );
    
        // Verificar que se renderizan las props personalizadas
        expect(screen.getByText('Custom Error')).toBeInTheDocument();
        expect(
          screen.getByText('This is a custom error description.')
        ).toBeInTheDocument();
      });
    
      test('no renderiza detalles si errorDetails no está presente', () => {
        render(<ErrorScreen />);
    
        // Verificar que los detalles no están en el documento
        expect(screen.queryByText('More details')).not.toBeInTheDocument();
      });
    
      test('renderiza los detalles del error cuando errorDetails está presente', () => {
        const errorDetails = 'Stack trace: line 42';
        render(<ErrorScreen errorDetails={errorDetails} />);
    
        // Verificar que los detalles están en el documento
        expect(screen.getByText('More details')).toBeInTheDocument();
        expect(screen.getByText(errorDetails)).toBeInTheDocument();
      });
    
      test('verifica que los atributos de accesibilidad estén configurados correctamente', () => {
        render(<ErrorScreen />);
    
        const alertSection = screen.getByRole('alert');
    
        // Verificar que los atributos ARIA están presentes y configurados correctamente
        expect(alertSection).toHaveAttribute('aria-live', 'assertive');
        expect(alertSection).toHaveAttribute('aria-atomic', 'true');
      });


});