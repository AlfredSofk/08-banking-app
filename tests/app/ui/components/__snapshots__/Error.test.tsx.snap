// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test para el componente ErrorScreen > match snapshot 1`] = `
<DocumentFragment>
  <section
    aria-atomic="true"
    aria-live="assertive"
    class="error-message"
    role="alert"
  >
    <header
      class="error-message__header"
    >
      <h1
        class="error-message__title"
      >
        Something went wrong
      </h1>
    </header>
    <div
      class="error-message__body"
    >
      <p
        class="error-message__description"
      >
        An unexpected error occurred. Please try again later.
      </p>
    </div>
  </section>
</DocumentFragment>
`;
